From a2b98bba2aad53f9c41ebdb9f3f27a252f6089db Mon Sep 17 00:00:00 2001
From: hhh <3224318821@qq.com>
Date: Sat, 21 Mar 2026 16:21:00 +0800
Subject: [PATCH 15/15] add uart_rx present


diff --git a/bsp/abstract-machine/extra.ld b/bsp/abstract-machine/extra.ld
index ef23176..f899bf7 100644
--- a/bsp/abstract-machine/extra.ld
+++ b/bsp/abstract-machine/extra.ld
@@ -29,7 +29,7 @@ SECTIONS {
     . = ALIGN(8);
   }
 }
-INSERT AFTER .data;
+INSERT BEFORE .data;
 
 SECTIONS {
   .bss.extra (NOLOAD) : {
@@ -42,4 +42,4 @@ SECTIONS {
     . = ALIGN(8);
   }
 }
-INSERT AFTER .bss;
+INSERT BEFORE .bss;
diff --git a/bsp/abstract-machine/src/uart.c b/bsp/abstract-machine/src/uart.c
index 2161e69..4fa0d0d 100644
--- a/bsp/abstract-machine/src/uart.c
+++ b/bsp/abstract-machine/src/uart.c
@@ -38,10 +38,8 @@ static int _uart_putc(struct rt_serial_device *serial, char c) {
 }
 
 static int _uart_getc(struct rt_serial_device *serial) {
-  char ch = io_read(AM_UART_RX).data; 
-  return (int)ch;
-  // static const char *p = "help\nam_hello\nam_microbench\n";
-  // return (*p != '\0' ? *(p ++) : io_read(AM_UART_RX).data);
+  static const char *p = "help\nam_hello\n";
+  return (*p != '\0' ? *(p ++) : io_read(AM_UART_CONFIG).present ? io_read(AM_UART_RX).data : -1);
 }
 
 const struct rt_uart_ops _uart_ops = {
-- 
2.34.1

